src/pretty/code_flow.rs

```diff
@@ -71,6 +71,10 @@ impl<'a> PrettyPrinter<'a> {
     }
 
     pub(super) fn convert_binary(&'a self, binary: Binary<'a>) -> ArenaDoc<'a> {
+        // Layout every binary expression except assignment as chain.
+        if binary.op().precedence() > 1 {
```

Comment:
nit: is it possible to check assign operator directly? IIUC assign operator's precedence happens to be 1. It would be better if we don't rely on that